artemis-commons/src/main/java/org/apache/activemq/artemis/utils/collections/LinkedListImpl.java

```diff
@@ -72,9 +124,41 @@ public void addHead(E e) {
          node.next.prev = node;
       }
 
+      itemAdded(node, e);
+
       size++;
    }
 
+   @Override
+   public E removeWithID(long id) {
+      if (nodeMap == null) {
+         return null;
+      }
+
+      Node<E> node = nodeMap.get(id);
+      if (node == null) {
+         return null;
+      }
+
+      removeAfter(node.prev);
+      return node.val();
+   }
+
+   private void itemAdded(Node node, E item) {
+      if (nodeMap != null) {
+         putID(item, node);
+      }
+   }
+
+   private void itemRemoved(Node node) {
+      if (nodeMap != null) {
+         Object id = idSupplier.applyAsLong((E)node.val());
+         if (id != null) {
+            nodeMap.remove(id);
```

Comment:
Node\<E\> to avoid type warnings, remove unncessary cast to E?

Change 'id' from Object to long, avoid boxing? Update or remove gating check as value cant be null.

Reply:
this is a leftover from the refactoring when the IDs were Objects.. thanks